Topic3/src/org/maugiam/tdd/main/WordWrapper.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * 
+ */
+package org.maugiam.tdd.main;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringTokenizer;
+
+/**
+ * @author Mauro J Giamberardino
+ *
+ */
+public class WordWrapper {
+
+	public static List<String> wrap(String string, Integer rowLength) {
+		ArrayList<String> res = new ArrayList<String>();
+		StringTokenizer tok = new StringTokenizer(string, " ");
+		if (string.length() <= rowLength)
+			res.add(string);
+		else {
+			String wordAux = "";
+			while (tok.hasMoreTokens()) {
+				String next = tok.nextToken();
+				if ((!wordAux.equals("")) && ((wordAux.length() + next.length() + 1) > rowLength)) {
+					res.add(wordAux);
+					wordAux = "";
+				}
+				if (next.length() <= rowLength) {
+					if (wordAux.length() > 0)
+						wordAux += " ";
+					wordAux += next;
+				} else {
+					String trim = next;
+					while (trim.length() > rowLength) {
+						res.add(trim.substring(0, rowLength));
+						trim = trim.substring(rowLength);
+					}
+					wordAux = trim;
+				}
+			}
+			if (!wordAux.equals(""))

```

Comment:
Warning: When using Strings it's a good practice to run the if the condition the other way round, i.e. !"".equals(wordAux). By doing this you avoid to run into a NullPointerException condition if wordAux was set as null.